### PR TITLE
[all] (Host-specific APIs) Create a host-specific API model page

### PR DESCRIPTION
**Pull request by AlexJerabek**
This page talks about the load/sync pattern found in the Excel, Word, OneNote, and Visio APIs. It mostly pulls information from two Excel articles ([here](https://docs.microsoft.com/en-us/office/dev/add-ins/excel/excel-add-ins-core-concepts) and [here](https://docs.microsoft.com/en-us/office/dev/add-ins/excel/excel-add-ins-advanced-concepts)) and puts it in a more top-level area.

I'll remove the duplicated information in the host sections in a subsequent PR.

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[b9d12b1](https://github.com/OfficeDev/office-js-docs-pr/commits/b9d12b114f16bc631bb5594677f234480a79a7a8)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/host-specific-api-model.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/develop/host-specific-api-model.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/develop/host-specific-api-model?branch=pr-en-us-2046) |
[docs/includes/office-js-api-models.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/includes/office-js-api-models.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins-fundamentals?branch=pr-en-us-2046) |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/toc.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/develop/host-specific-api-model?branch=pr-en-us-2046) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C7%5C27%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202007272330525837-2046%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=xhUPrt7OrEQpuzoWxNFTaUS6gP0SZRdZf%2FGxBWbtI0Y%3D&st=2020-07-27T23%3A27%3A42Z&se=2020-08-27T23%3A32%3A42Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/57856ef7-e8cf-54ca-0d12-df5f42d1e8bb?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

File: docs/develop/host-specific-api-model.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+## Proxy objects
+
+The Office JavaScript objects that you declare and use with the promise-based APIs are proxy objects. Any methods that you invoke or properties that you set or load on proxy objects are simply added to a queue of pending commands. When you call the `sync()` method on the request context (for example, `context.sync()`), the queued commands are dispatched to the Office host and run. These APIs are fundamentally batch-centric. You can queue up as many changes as you wish on the request context, and then call the `sync()` method to run the batch of queued commands.
+
+For example, the following code snippet declares the local JavaScript [Excel.Range](/javascript/api/excel/excel.range) object, `selectedRange`, to reference a selected range in the Excel workbook, and then sets some properties on that object. The `selectedRange` object is a proxy object, so the properties that are set and method that is invoked on that object will not be reflected in the Excel document until your add-in calls `context.sync()`.
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/host-specific-api-model.md:47]:
```suggestion
For example, the following code snippet declares the local JavaScript [Excel.Range](/javascript/api/excel/excel.range) object, `selectedRange`, to reference a selected range in the Excel workbook, and then sets some properties on that object. The `selectedRange` object is a proxy object, so the properties that are set and the method that is invoked on that object will not be reflected in the Excel document until your add-in calls `context.sync()`.
```

File: docs/develop/host-specific-api-model.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+}).catch(function (error) {
+    console.log('Error: ' + error);
+    if (error instanceof OfficeExtension.Error) {
+        console.log('Debug info: ' + JSON.stringify(error.debugInfo));
+    }
+});
+```
+
+> [!NOTE]
+> If you are only calling methods or setting properties on a proxy object, you do not need to call the `load()` method. The `load()` method is only required when you want to read properties on a proxy object.
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/host-specific-api-model.md:110]:
```suggestion
> If you are only calling methods or setting properties on a proxy object, you don't need to call the `load()` method. The `load()` method is only required when you want to read properties on a proxy object.
```

File: docs/develop/host-specific-api-model.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+The following code sample attempts to retrieve an Excel worksheet named "Data" by using the `getItemOrNullObject()` method. If the method returns a null object, a new sheet needs to be created before actions can taken on the sheet.
+
+```js
+var dataSheet = context.workbook.worksheets.getItemOrNullObject("Data");
+
+return context.sync()
+  .then(function() {
+    if (dataSheet.isNullObject) {
+        // Create the sheet
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/host-specific-api-model.md:201]:
```suggestion
        // Create the sheet.
```

File: docs/includes/office-js-api-models.md [hunk 4]
```diff
@@ -13,6 +13,8 @@ The Office JavaScript API includes two distinct models:
     > [!NOTE]
     > There is also a host-specific API for [Visio](../reference/overview/visio-javascript-reference-overview.md), but you can use it only in SharePoint Online pages to interact with Visio diagrams that have been embedded in the page. Office web add-ins are not supported in Visio.
 
+    Learn more about how to use this API model in the article [Using the host-specific API model](../develop/host-specific-api-model.md).
```
Comment (ElizabethSamuel-MSFT) [inline docs/includes/office-js-api-models.md:16]:
```suggestion
    See [Using the host-specific API model](../develop/host-specific-api-model.md) to learn more about this API model .
```

File: docs/develop/host-specific-api-model.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+> [!NOTE]
+> The examples in this page use the Excel JavaScriptAPIs, but the concepts equally apply to Excel, Word, and OneNote add-ins, as well as SharePoint-embedded Visio diagrams that use the Office JavaScript APIs.
+
+## Asynchronous nature of the promise-based APIs
+
+The web-based add-ins run inside a browser container. This container is embedded within the Office application on desktop-based platforms, such as Office on Windows, and run inside an HTML iFrame in Office on the web. Enabling the Office.js API to interact synchronously with the Office host across all supported platforms is not feasible due to performance considerations. Therefore, the `sync()` API call in Office.js returns a [promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise) that is resolved when the Office application completes the requested read or write actions. Also, you can queue up multiple actions, such as setting properties or invoking methods, and run them as a batch of commands with a single call to `sync()`, rather than sending a separate request for each action. The following sections describe how to accomplish this using the `run()` and `sync()` APIs.
```
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:17]: I can't tell what the change is here.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[2ee109c](https://github.com/OfficeDev/office-js-docs-pr/commits/2ee109c8faa34408c24e154c56a4857fda1311b7)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/host-specific-api-model.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/develop/host-specific-api-model.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/develop/host-specific-api-model?branch=pr-en-us-2046) |
[docs/includes/office-js-api-models.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/includes/office-js-api-models.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins-fundamentals?branch=pr-en-us-2046) |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/toc.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/develop/host-specific-api-model?branch=pr-en-us-2046) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C7%5C28%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202007281735236965-2046%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=Cgl9YTfTe%2B5Tj%2FM4DROn8McsU6OjTRmIodkwIX0OCYE%3D&st=2020-07-28T17%3A32%3A28Z&se=2020-08-28T17%3A37%3A28Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/57856ef7-e8cf-54ca-0d12-df5f42d1e8bb?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

File: docs/develop/host-specific-api-model.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ElizabethSamuel-MSFT) [reply docs/develop/host-specific-api-model.md:17]: I suggested you change `...and run inside an HTML...` to `...and runs inside an HTML...`.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[223dea7](https://github.com/OfficeDev/office-js-docs-pr/commits/223dea7d6b678ff61d49600366a5efc19962fd9a)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/host-specific-api-model.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/develop/host-specific-api-model.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/develop/host-specific-api-model?branch=pr-en-us-2046) |
[docs/includes/office-js-api-models.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/includes/office-js-api-models.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins-fundamentals?branch=pr-en-us-2046) |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/toc.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/develop/host-specific-api-model?branch=pr-en-us-2046) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C7%5C28%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202007281735510359-2046%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=o3nd4LauolWXzzcSDO16nHqtla%2Fc8uwcDhQkWIhbhV8%3D&st=2020-07-28T17%3A34%3A24Z&se=2020-08-28T17%3A39%3A24Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/57856ef7-e8cf-54ca-0d12-df5f42d1e8bb?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

File: docs/develop/host-specific-api-model.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:17]: Thanks! Not sure why GitHub didn't highlight that...

**Comment by AlexJerabek** [comment]
Thanks for the review @ElizabethSamuel-MSFT! I've applied all your changes. Is this good to be checked in?

**Comment by o365devx** [comment]
Docs Build status updates of commit _[75c1e0e](https://github.com/OfficeDev/office-js-docs-pr/commits/75c1e0e0ff67624dd320c3afea48c5a620f9bf47)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/host-specific-api-model.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/develop/host-specific-api-model.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/develop/host-specific-api-model?branch=pr-en-us-2046) |
[docs/includes/office-js-api-models.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/includes/office-js-api-models.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins-fundamentals?branch=pr-en-us-2046) |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-ExcelSpecificMaterial/docs/toc.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/develop/host-specific-api-model?branch=pr-en-us-2046) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C7%5C28%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202007281741399973-2046%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=TdzRcUUwtKj2%2Fxw6rglgcEzg3rNC44GatfeOIpkhUAA%3D&st=2020-07-28T17%3A38%3A44Z&se=2020-08-28T17%3A43%3A44Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/57856ef7-e8cf-54ca-0d12-df5f42d1e8bb?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

File: docs/develop/host-specific-api-model.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+> [!NOTE]
+> The examples in this page use the Excel JavaScriptAPIs, but the concepts equally apply to Excel, Word, and OneNote add-ins, as well as SharePoint-embedded Visio diagrams that use the Office JavaScript APIs.
+
+## Asynchronous nature of the promise-based APIs
+
+The web-based add-ins run inside a browser container. This container is embedded within the Office application on desktop-based platforms, such as Office on Windows, and runs inside an HTML iFrame in Office on the web. Enabling the Office.js API to interact synchronously with the Office host across all supported platforms is not feasible due to performance considerations. Therefore, the `sync()` API call in Office.js returns a [promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise) that is resolved when the Office application completes the requested read or write actions. Also, you can queue up multiple actions, such as setting properties or invoking methods, and run them as a batch of commands with a single call to `sync()`, rather than sending a separate request for each action. The following sections describe how to accomplish this using the `run()` and `sync()` APIs.
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:17]: I don't know where the suggestion button went but here is a suggestion: 
Office Add-ins are websites which appear inside a browser container within Office hosts, such as Excel.
Then leave everything through Office on the web. Then a space. 
Due to performance considerations, the Office.js APIs cannot interact synchronously with the Office hosts across all platforms. 
Then the rest looks good

File: docs/develop/host-specific-api-model.md [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+## *.run function
+
+`Excel.run`, `Word.run`, and `OneNote.run` execute a function that specifies the actions to perform against Excel, Word, and OneNote. `*.run` automatically creates a request context that you can use to interact with Office objects. When `*.run` completes, a promise is resolved, and any objects that were allocated at runtime are automatically released.
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:21]: So I don't know what the team's thoughts are globally on this but do we write Promise or promise? Technically it is the caps, but I think the lower case makes more sense

File: docs/develop/host-specific-api-model.md [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+
+The following example shows how to use `Excel.run`. The same pattern is also used with Word and OneNote.
+
+```js
+Excel.run(function (context) {
+    // You can use the Excel JavaScript API here in the batch function
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:27]: Maybe 
Batch Excel JS API calls here that will execute actions on the Excel Object model

File: docs/develop/host-specific-api-model.md [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+selectedRange.format.fill.color = "#4472C4";
+selectedRange.format.font.color = "white";
+selectedRange.format.autofitColumns();
+```
+
+### sync()
+
+Calling the `sync()` method on the request context synchronizes the state between proxy objects and objects in the Office document. The `sync()` method runs any commands that are queued on the request context and retrieves values for any properties that should be loaded on the proxy objects. The `sync()` method executes asynchronously and returns a [promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise), which is resolved when the `sync()` method completes.
+
+The following example shows a batch function that defines a local JavaScript proxy object (`selectedRange`), loads a property of that object, and then uses the JavaScript Promises pattern to call `context.sync()` to synchronize the state between proxy objects and objects in the Excel document.
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:60]: Here promises is caps

File: docs/develop/host-specific-api-model.md [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+    console.log('error: ' + error);
+    if (error instanceof OfficeExtension.Error) {
+        console.log('Debug info: ' + JSON.stringify(error.debugInfo));
+    }
+});
+```
+
+In the previous example, `selectedRange` is set and its `address` property is loaded when `context.sync()` is called.
+
+Because `sync()` is an asynchronous operation that returns a promise, you should always `return` the promise (in JavaScript). Doing so ensures that the `sync()` operation completes before the script continues to run. For more information about optimizing performance with `sync()`, see [Excel JavaScript API performance optimization](../excel/performance.md).
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:80]: Maybe 
Because sync is an asynchronous operation, always return the promise to ensure the sync operation completes before the script runs

File: docs/develop/host-specific-api-model.md [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+```
+
+> [!NOTE]
+> If you are only calling methods or setting properties on a proxy object, you don't need to call the `load()` method. The `load()` method is only required when you want to read properties on a proxy object.
+
+Just like requests to set properties or invoke methods on proxy objects, requests to load properties on proxy objects get added to the queue of pending commands on the request context, which will run the next time you call the `sync()` method. You can queue up as many `load()` calls on the request context as necessary.
+
+#### Scalar and navigation properties
+
+There are two categories of properties: **scalar** and **navigational**. Scalar properties are assignable types such as strings, integers, and JSON structs. Navigation properties are readonly objects and collections of objects that have their fields assigned, instead of directly assigning the property. For example, `name` and `position` members on the [Excel.Worksheet](/javascript/api/excel/excel.worksheet) object are scalar properties, whereas `protection` and `tables` are navigation properties.
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:116]: I think read-only needs the -

File: docs/develop/host-specific-api-model.md [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+
+```js
+someRange.load("format/font/name")
+```
+
+You can also set the scalar properties of a navigation property by traversing the path. For example, you could set the font size for an `Excel.Range` by using `someRange.format.font.size = 10;`. You don't need to load the property before you set it.
+
+#### Calling `load` without parameters
+
+If you call the `load()` method on an object (or collection) without specifying any parameters, all scalar properties of the object (or all scalar properties of all objects in the collection) will be loaded. To reduce the amount of data transfer between the Excel host application and the add-in, you should avoid calling the `load()` method without explicitly specifying which properties to load.
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:128]: Maybe 
all scalar props of the object or the collection's objects will be loaded

File: docs/develop/host-specific-api-model.md [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+
+> [!IMPORTANT]
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:130]: I dunno this seems Excel specific?

File: docs/develop/host-specific-api-model.md [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+    console.log("Error: " + error);
+    if (error instanceof OfficeExtension.Error) {
+        console.log("Debug info: " + JSON.stringify(error.debugInfo));
+    }
+});
+```
+
+## get&#42;OrNullObject methods
+
+Some `get*` methods throw an exception when the desired object is not present. For example, if you attempt to get an Excel worksheet by specifying a worksheet name that is not in the workbook, the `getItem()` method throws an `ItemNotFound` exception.
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:189]: isn't
Sorry I forgot to check for contractions throughout, you may want to have a cursory look

File: docs/develop/host-specific-api-model.md [hunk 15]
```diff
@@ -0,0 +1,211 @@
...
+
+Any `get*OrNullObject` method variant lets you check for an object without throwing exceptions. These methods return a null object (not the JavaScript `null`) rather than throwing an exception if the specified item doesn't exist. For example, you can call the `getItemOrNullObject()` method on a collection such as **Worksheets** to attempt to retrieve an item from the collection. The `getItemOrNullObject()` method returns the specified item if it exists; otherwise, it returns a null object. The null object that is returned contains the boolean property `isNullObject` that you can evaluate to determine whether the object exists.
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:191]: I think you can get rid of attempt to, i.e.
**Worksheets** to retrieve an item...

File: docs/develop/host-specific-api-model.md [hunk 16]
```diff
@@ -0,0 +1,211 @@
...
+```js
+var dataSheet = context.workbook.worksheets.getItemOrNullObject("Data");
+
+return context.sync()
+  .then(function() {
+    if (dataSheet.isNullObject) {
+        // Create the sheet.
+    }
+
+    dataSheet.position = 1;
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:204]: Do you need this?

File: docs/develop/host-specific-api-model.md [hunk 17]
```diff
@@ -0,0 +1,211 @@
...
+    //...
+  })
+```
+
```
Comment (mscharlock) [inline docs/develop/host-specific-api-model.md:208]: Just putting a random comment here: you know what might be nice is a flow chart/graphic at the top of this page for people like 
1. Run 
2. Load
etc

**Comment by mscharlock** [comment]
Also I wonder where this should live in the TOC. It makes sense with our current organization where you have put it now, but I almost feel like it needs to be called out on the Overview page as a link under "Concept" and also in that "Get Started" section, since it's pretty essential info. Right now it is kinda nested away

File: docs/develop/host-specific-api-model.md [hunk 18]
```diff
@@ -0,0 +1,211 @@
+---
+title: Using the host-specific API model
```
[the remark this answers is not in the document]
Comment (alison-mk) [reply docs/develop/host-specific-api-model.md:2]: I have the same question. As a newcomer, I find seeing 'host' in the article title confusing. My understanding is that host refers to the potential applications one can develop add-ins for (OneNote, Excel, etc), but this was not immediately clear to me when reading the article title and I'm actually still not sure if that's what host means in this context. I think 'host' is a broad word with a lot of potential meanings to developers. Could the title be more specific?

[the remark this answers is not in the document]
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:2]: I'm open to naming suggestions. There's a definite need for a term to describe what we internally call "Rich APIs". Existing articles use "Excel JavaScript APIs", "Word JavaScript APIs", and "OneNote JavaScript APIs" in their respective sections, so I figured that generalizes to "host-specific" APIs. There were also some articles using the "host-specific" term that Kim had added.

My broader hope is that by the end of this refactoring, whatever term we settle on becomes prominent enough to become the obvious alternative to "Common APIs". That way, users see the choice as "do I want to learn about APIs that are common to all hosts or just used by a specific host."

File: docs/develop/host-specific-api-model.md [hunk 19]
```diff
@@ -0,0 +1,211 @@
...
+ms.date: 07/27/2020
+localization_priority: Normal
+---
+
+# Using the host-specific API model
+
+This article describes how to use the API model for building add-ins in Excel, Word, and OneNote. It introduces core concepts that are fundamental to using the promise-based APIs. Note that this model is not supported by Office 2013 clients. Use the [Common API model](office-javascript-api-object-model.md) to work with those Office versions. For full platform availability notes, see [Office Add-in host and platform availability](../overview/office-add-in-availability.md).
+
+> [!NOTE]
+> The examples in this page use the Excel JavaScriptAPIs, but the concepts equally apply to Excel, Word, and OneNote add-ins, as well as SharePoint-embedded Visio diagrams that use the Office JavaScript APIs.
```
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:13]: Fixed the space. I really want both parts to be a note, but that looks bad, right?

File: docs/develop/host-specific-api-model.md [hunk 20]
```diff
@@ -0,0 +1,211 @@
+---
+title: Using the host-specific API model
+description: 'Learn about the promise-based API model for Excel, OneNote, and Word add-ins.'
+ms.date: 07/27/2020
+localization_priority: Normal
+---
+
+# Using the host-specific API model
+
+This article describes how to use the API model for building add-ins in Excel, Word, and OneNote. It introduces core concepts that are fundamental to using the promise-based APIs. Note that this model is not supported by Office 2013 clients. Use the [Common API model](office-javascript-api-object-model.md) to work with those Office versions. For full platform availability notes, see [Office Add-in host and platform availability](../overview/office-add-in-availability.md).
```
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:10]: I'll reword it to avoid the mention.

File: docs/develop/host-specific-api-model.md (hunk 7 above)
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:21]: My rule of thumb is that whenever you're talking about the concept, it's lowercase. If you're talking about the code object, it's `Promise`. It's the same logic I use for range versus `Range`.

File: docs/develop/host-specific-api-model.md (hunk 8 above)
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:27]: How about "// Add your Excel JS API calls here that will be batched and sent to the workbook."

I almost don't want to mention batching here, since you can have several batches in the `run` call.

File: docs/develop/host-specific-api-model.md (hunk 10 above)
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:80]: Good call. I also added a bit about using `await` in TS/ES6+ JS.

File: docs/develop/host-specific-api-model.md (hunk 13 above)
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:130]: Agreed. I'm going to leave the section of exceptions there because I don't have another place to put it yet.

File: docs/develop/host-specific-api-model.md (hunk 16 above)
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:204]: I think it highlights the pattern of "if null: create the thing, then do the operation regardless".

File: docs/develop/host-specific-api-model.md (hunk 17 above)
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:208]: We need a good drawing of how proxy objects work in general... I'll add it to the refactoring list.

**Comment by AlexJerabek** [comment]
Thanks for the review @mscharlock. I've added the updates to #2050.

File: docs/develop/host-specific-api-model.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (ElizabethSamuel-MSFT) [reply docs/develop/host-specific-api-model.md:2]: Actually, maybe using a term like "Office hosts" in the L2 will help with this. Also, there are "Host" and "Hosts" elements in the manifest that represent which Office hosts your add-in supports. We don't have to use the manifest term but it's a term the user will need to become somewhat familiar with.

File: docs/develop/host-specific-api-model.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (ElizabethSamuel-MSFT) [reply docs/develop/host-specific-api-model.md:13]: Maybe make one "IMPORTANT" and the other a "NOTE"

File: docs/develop/host-specific-api-model.md (hunk 13 above)
Comment (ElizabethSamuel-MSFT) [reply docs/develop/host-specific-api-model.md:130]: Are we aware of any similar exceptions for non-Excel?

File: docs/develop/host-specific-api-model.md (hunk 16 above)
Comment (ElizabethSamuel-MSFT) [reply docs/develop/host-specific-api-model.md:204]: Might be useful to add a comment to that effect here... e.g., "Do operation on object."

File: docs/develop/host-specific-api-model.md (hunk 13 above)
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:130]: @ElizabethSamuel-MSFT, Excel is the only one with a different `load` behavior because of the sheer number of properties in `Range` (and some customers load large Ranges through Graph).

File: docs/develop/host-specific-api-model.md (hunk 16 above)
Comment (AlexJerabek) [reply docs/develop/host-specific-api-model.md:204]: I'll do you one better and make the sample actually functional :-)
